### PR TITLE
/tg/ Status Effects Prelude A - Updated Trait Backend

### PR DESCRIPTION
**Pull request by fira**
# About the pull request

Content from #4828 broken down in an individual PR

This does only a few things:
 * Adds /tg/ ASSERT which includes reasons
 * Updates the Traits backend from /tg/
 * Refactors cloak cancellation with signals because it used REMOVE_TRAIT_ALLSOURCES which is both bad and defunct

# Testing Photographs and Procedure
Tested trait backend as part of main PR

### CODE REVIEW

File: code/game/objects/items/handcuffs.dm [hunk 1]
```diff
@@ -29,20 +29,6 @@
 	if(!C.handcuffed)
 		place_handcuffs(C, user)
 
-/obj/item/handcuffs/obj/structure/MouseDrop(mob/living/carbon/human/H)
```
Comment (Zonespace27) [inline code/game/objects/items/handcuffs.dm:32]: cursed

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -513,6 +513,32 @@
 
 	return do_pull(AM, lunge, no_msg)
 
+/mob/proc/stop_pulling()
+	if(pulling)
```
Comment (Zonespace27) [inline code/modules/mob/mob.dm:517]: early return

File: code/modules/mob/mob.dm [hunk 3]
```diff
@@ -513,6 +513,32 @@
...
+		var/mob/M = pulling
```
Comment (Zonespace27) [inline code/modules/mob/mob.dm:518]: one letter var

File: code/modules/mob/mob.dm [hunk 4]
```diff
@@ -513,6 +513,32 @@
 
 	return do_pull(AM, lunge, no_msg)
 
+/mob/proc/stop_pulling()
```
Comment (Zonespace27) [inline code/modules/mob/mob.dm:516]: code doc

File: code/modules/mob/mob.dm [hunk 5]
```diff
@@ -513,6 +513,32 @@
...
+		pulling.pulledby = null
+		pulling = null
+
+		grab_level = 0
+		if(client)
+			client.recalculate_move_delay()
+			// When you stop pulling a mob after you move a tile with it your next movement will still include
+			// the grab delay so we have to fix it here (we love code)
+			client.next_movement = world.time + client.move_delay
+		if(hud_used && hud_used.pull_icon)
```
Comment (Zonespace27) [inline code/modules/mob/mob.dm:528]:
```suggestion
		if(hud_used?.pull_icon)
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/game/objects/items/handcuffs.dm (hunk 1 above)
Comment (fira) [reply code/game/objects/items/handcuffs.dm:32]: It is, that's why i'm getting rid of it on the spot

**Review by Zonespace27 — APPROVED** [review]

**Pull request merged**